ray-curator/ray_curator/stages/image/filters/nsfw_filter.py

```diff
@@ -0,0 +1,116 @@
+from dataclasses import dataclass
+
+import numpy as np
+import torch
+from loguru import logger
+
+from ray_curator.backends.base import WorkerMetadata
+from ray_curator.models.nsfw import NSFWScorer
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.resources import Resources
+from ray_curator.tasks import ImageBatch
+
+
+@dataclass
+class ImageNSFWFilterStage(ProcessingStage[ImageBatch, ImageBatch]):
+    """Stage for filtering out NSFW images using NSFWScorer model.
+
+    This class processes image batches through an NSFW scoring model to generate
+    NSFW probability scores for each image. Images with scores above the threshold
+    will be filtered out as NSFW content.
+    """
+    model_dir: str = "models/nsfw"
+    num_gpus_per_worker: float = 0.25
+    model_batch_size: int = 32  # Number of images to process through model at once
```

Comment:
```suggestion
    model_inference_batch_size: int = 32  # Number of images to process through model at once
```
Same comment as above.